addon/ajax-request.js

```diff
@@ -42,6 +42,35 @@ function isJSONAPIContentType(header) {
   return header.indexOf(JSONAPIContentType) === 0;
 }
 
+function startsWithSlash(string) {
+  if (typeof string === 'string') {
+    return string.charAt(0) === '/';
+  }
+  return string;
+}
+
+function endsWithSlash(string) {
+  if (typeof string === 'string') {
+    return string.charAt(string.length - 1) === '/';
+  }
+  return string;
+}
+
+function stripSlashes(path) {
+  if (typeof path === 'string') {
```

Comment:
No, I'm referring to `typeof path === 'string'`. You don't need to check the argument type here.

It's safe enough for stripSlashes to either force `path` argument to be a string or throw an error here if it's not.